aiida/orm/nodes/data/array/cube.py

```diff
@@ -0,0 +1,246 @@
+
+    @property
+    def data(self):
+        """3D array containing the mesh data."""
+        return self.get_array('data')
+
+    @data.setter
+    def data(self, data):
+        self.set_array('data', data)
+
+    @property
+    def data_units(self):
+        """Units of the mesh data."""
+        return self.get_attribute('data_units')
+
+    @data_units.setter
+    def data_units(self, data_units):
+        self.set_attribute('data_units', data_units)
+
+    def read(self, file, data_units=None):
+        """Parse cube file."""
+
+        # It should be either a string pointing to a file, or a file object.
+        if isinstance(file, str):
+            fiter = iter(open(file).readlines())
+        else:
+            fiter = iter(file.readlines())
+
+        # Comment.
+        self.comment = [next(fiter).rstrip(), next(fiter).rstrip()]
+
+        # Number of atoms and origin.
+        line = next(fiter).split()
+        natoms = int(line[0])
+        self.origin = np.array(line[1:], dtype=float)
+
+        # Data shape and voxel.
+        shape = np.empty(3, dtype=int)
+        voxel = np.empty((3, 3))
+        for i in range(3):
+            nmb, x, y, z = [float(s) for s in next(fiter).split()]
+            shape[i] = int(nmb)
+            voxel[i] = np.array([x, y, z])
+
+        self.voxel = voxel
+
+        # Atomic numbers, charges and positions.
+        numbers = np.empty(natoms, int)
+        charges = np.empty(natoms, int)
+        positions = np.empty((natoms, 3))
+        for i in range(natoms):
+            line = next(fiter).split()
+            numbers[i] = int(line[0])
+            charges[i] = float(line[1])
+            positions[i] = [float(s) for s in line[2:]]
+        self.atomic_numbers = numbers
+        self.atomic_charges = charges
+        self.atomic_coordinates = positions
+
+        # Data mesh.
+        data = np.empty(shape[0] * shape[1] * shape[2], dtype=float)
+        cursor = 0
+        for i, line in enumerate(fiter):
+            l_split = line.split()
+            data[cursor:cursor + len(l_split)] = l_split
+            cursor += len(l_split)
+        self.data = data.reshape(shape)
+
+        if data_units:
+            self.data_units = data_units
+
+    def everything_ok(self):
+        """Check if the object contains all the data to construct a valid cube file."""
+        all_good = True
+        for attr in ['comment', 'voxel', 'origin', 'atomic_numbers', 'atomic_charges', 'atomic_coordinates', 'data']:
+            try:
+                getattr(self, attr)
+            except (AttributeError, KeyError):
+                all_good = False
+                print(f'The attribute `{attr}` is not set.')
+        return all_good
+
+    def export(self, path, fileformat='cube', overwrite=False, **kwargs):
+        """Export data to the selected file."""
+
+        if not self.everything_ok():
+            raise ValueError("Can't export the cube file.")
```

Comment:
Again, don't really see the point of this. There are a number of ways wherein this test will pass, but the `_write_gaussian_cube` will fail. Why do you want to have a separate check that only catches a small portion of cases?